src/tree/c14n.rs

```diff
@@ -0,0 +1,44 @@
+//! Shared canonicalization logic and types.
+//!
+use std::ffi::c_int;
+
+use crate::bindings::{
+  xmlC14NMode_XML_C14N_1_0, xmlC14NMode_XML_C14N_1_1, xmlC14NMode_XML_C14N_EXCLUSIVE_1_0,
+};
+
+/// Options for configuring how to canonicalize XML
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Default)]
+pub struct CanonicalizationOptions {
```

Comment:
With `inclusive_ns_prefixes` as part of the `CanonicalizationMode` (see below), there are only two options left. For those, I'd rather have them as arguments to the functions instead of having an options struct.